### ISSUE TITLE
VS Code cannot find "dotenv/types"

### ISSUE THREAD

**Issue by alexesprit** [motdotla/dotenv#475]
**TL; DR:** VS Code displays an error it can't find `dotenv/types`, this can be fixed by replacing the `types` property of the `package.json` file with "types/index.d.ts".

I have a project written in JavaScript, where `dotenv` is used as a dev dependency. I added a `jsconfig.json` file to the project recently, and noticed an error VS Code displayed for this file:
```
File '%path_to_project%/node_modules/dotenv/types' not found.
```
<details>
  <summary>Screenshot (Click to expand)</summary>

Content
![image](https://user-images.githubusercontent.com/1119267/86647333-a2a10180-bfe8-11ea-8192-87adf082b054.png)
</details>

I found out the `types` property of dotenv's `package.json` file is set to "types" which is directory where `index.d.ts` is stored. For some reason VS Code `ts` checker treats "types" as a file, and cannot find it, as it does not exist.

I changed this property to "types/index.d.ts" in the `package.json` right in my `node_modules` directory. I see no errors in VS Code after this change. If this fix is acceptable I can make a PR with this change.

Also, in TypeScript documentation is said the `types` property should point to a declaration file, not a directory.
Ref https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package.

Interesting that seems like VS Code is able to load the `index.d.ts` file in some cases (e.g. it displays documentation and types for dotenv functions in the code editor), even if I keep `types` property untouched. Not sure why it does work in the editor, and why it displays error for the `jsconfig.json` file at the same time.

**Comment by zenflow** [comment]
@alexesprit Yeah, the "types" property in package.json is a file (not a directory) in every project I've ever seen. 

I would go ahead and make a one-liner PR if I was you. Cheers!

**Comment by riyadshauk** [comment]
This fix doesn't seem to work for me... Did I miss something? I'm just doing the standard `// @ts-ignore` over my import of `dotenv` for now. Conveniently, this isn't too problematic for me since I don't really use `dotenv` in my code much, anyways.

**Comment by alexesprit** [comment]
@riyadshauk not sure why it does not work for you. I guess you restarted VS Code after applying changes, or closed it before applying changes.

By the way, thank you for the `ts-ignore` tip! I'll try this way.

**Comment by Anoesj** [comment]
Restarting VSCode fixed this, thanks. Weird bug though..

**Comment by pierlucg-xs** [comment]
Restarting VSCode fixes the issue but it comes back once in a while and you'll need to restart again, so it's only an (annoying) workaround for now.

**Comment by SeedBoot** [comment]
While not an exact solution, I tend to use dotenv as part of a `package.json` script e.g. `nodemon -r dotenv/config`. No errors so far 🤷

**Comment by lamuertepeluda** [comment]
Thanks @SeedBoot 

You pointed me in the right direction: removing the require and using dotenv/config solved the issue for me as well!

For those of you out there having this kind of trouble when using `jest`, here is my solution:

In package.json

```json
"scripts": {
    ...
    "test": "jest <pattern> dotenv_config_path=<env_file> ...", 
    ...
```

es. `jest test/** dotenv_config_path=.env.test --runInBand --detectOpenHandles`

in `jest.config.js` (or whatever config file format you like)


```js
// require('dotenv').config(...) // this was causing troubles, remove it!

module.exports = {
  ...
  setupFiles: [
    'dotenv/config',
    'anyother.script.js', // any other initialization script that relies on dotenv loaded properly -  omit if you don't have any
  ],
  ...
   }
}
```

Perhaps not the best solution, but at least removes the annoying error message in VS Code

**Comment by lamuertepeluda** [comment]
Sadly the issue suddenly reappeared, even though I am not requiring dotenv explicitly anymore. 😞  Very annoying!

**Comment by motdotla** [comment]
this will be released here: https://github.com/motdotla/dotenv/pull/519
